pkg/manifests/external_dns.go

```diff
@@ -86,11 +78,11 @@ func (p Provider) string() string {
 }
 
 type InputExternalDNSConfig struct {
-	TenantId, Subscription, ResourceGroup, ClientId, InputServiceAccount, Namespace, InputResourceName string
-	IdentityType                                                                                       IdentityType
-	ResourceTypes                                                                                      []ResourceType
-	Provider                                                                                           Provider
-	DnsZoneresourceIDs                                                                                 []string
+	TenantId, ClientId, InputServiceAccount, Namespace, InputResourceName string
+	Provider                                                              *Provider
```

Comment:
why is provider changing to pointer?

Reply:
seems like you're mixing responsibilities of crd controller with this package. They're different levels of abstraction